### DISCUSSION TITLE
Tailwind doesn't support transition on text-size?

### DISCUSSION THREAD

**Discussion by tarik-besic** [tailwindlabs/tailwindcss#10303] [Help] [answered]
In my Nextjs App I have this case
So I have a label and it has the default text-lg but then conditionally it gets the text-xs but the transition doesn't happen on the text-size but on other properties it does? (Using transition-all)

```
<label
onClick={(e) => {
setActive(true);
inputRef.current!.focus();
}}
className={`absolute cursor-pointer text-neutral-500 text-lg transition-all duration-100 ease-in-out ${
transformInput ? "text-xs -translate-y-2" : ""
}`}
>
{label}
</label>
```

**Comment by wongjn** [comment, marked as answer]
You'd probably want to only ensure one of the `text-*` classes is applied at one time, since one cannot guarantee the source order of the class names in the generated CSS:
```diff
 <label
 onClick={(e) => {
 setActive(true);
 inputRef.current!.focus();
 }}
-className={`absolute cursor-pointer text-neutral-500 text-lg transition-all duration-100 ease-in-out ${
-transformInput ? "text-xs -translate-y-2" : ""
+className={`absolute cursor-pointer text-neutral-500 transition-all duration-100 ease-in-out ${
+transformInput ? "text-xs -translate-y-2" : "text-lg"
 }`}
 >
 {label}
 </label>
```

**Reply by tarik-besic** [reply]
That's it, This helped me out a lot. You were right, **one cannot guarantee the source order of the class names in the generated CSS so always having one of the `text-*` classes solved the problem**
